vm/systemSmartContracts/governance.go

```diff
@@ -557,9 +560,16 @@ func (g *governanceContract) updateUserVoteList(address []byte, nonce uint64, di
 			return err
 		}
 	} else {
-		userVoteList.Delegated, err = addNewNonce(userVoteList.Delegated, nonce)
-		if err != nil {
-			return err
+		if !g.enableEpochsHandler.IsFlagEnabled(common.GovernanceFixesFlag) {
```

Comment:
Please refactor this by using several smaller functions. if else on an if else is hard to read